test/test_utils.jl

```diff
@@ -143,6 +144,49 @@ function test_jacobians(backend; multiple_inputs=true, test_types=true)
     @test yvec == yvec2
 end
 
+function test_second_derivatives(backend; multiple_inputs=false, test_types=true)
+    if multiple_inputs
+        # ... but 
+        error("multiple_inputs=true is not supported.")
+    else
+        # explicit test that AbstractDifferentiation throws an error
+        # don't support tuple of second derivatives
+        @test_throws ArgumentError AD.second_derivative(
+            backend, x -> fder(x, yscalar), (xscalar, yscalar)
+        )
+        @test_throws MethodError AD.second_derivative(
+            backend, x -> fder(x, yscalar), xscalar, yscalar
+        )
+    end
+
+    # test if single input (no tuple works)
+    dder1 = AD.second_derivative(backend, x -> fder(x, yscalar), xscalar)
+    if test_types
+        @test dder1[1] isa Float64
+    end
+    @test dfderdxdx(xscalar, yscalar) ≈ dder1[1] atol = 1e-8
+    valscalar, dder2 = AD.value_and_second_derivative(
+        backend, x -> fder(x, yscalar), xscalar
+    )
+    if test_types
+        @test valscalar isa Float64
+        @test dder2[1] isa Float64
+    end
+    @test valscalar == fder(xscalar, yscalar)
+    @test norm.(dder2 .- dder1) == (0,)
```

Comment:
Isn't `norm` unnecessarily complicated here?
```suggestion
    @test dder2 == dder1
```